modules/frontend/searchsharding.go

```diff
@@ -367,11 +372,16 @@ func (s *searchSharder) backendRequests(ctx context.Context, tenantID string, pa
 	return reqs, nil
 }
 
-// queryIngesterWithin returns a new start and end time range for the backend as well as an http request
+// ingesterRequest returns a new start and end time range for the backend as well as an http request
 // that covers the ingesters. If nil is returned for the http.Request then there is no ingesters query.
 // since this function modifies searchReq.Start and End we are taking a value instead of a pointer to prevent it from
 // unexpectedly changing the passed searchReq.
 func (s *searchSharder) ingesterRequest(ctx context.Context, tenantID string, parent *http.Request, searchReq tempopb.SearchRequest) (*http.Request, error) {
+	// request without start or end, search only in ingester
+	if searchReq.Start == 0 || searchReq.End == 0 {
```

Comment:
i like this fix. i think it nicely combines the old idea of "SearchRecent" with existing search.